# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: wheels
  joint_model_group_pose: home
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - front_left_wheel
    - front_right_wheel
    - rear_left_wheel
    - rear_right_wheel
  sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50


jaguar_robot:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Position Controllers ---------------------------------------
  joint1_position_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_left_wheel
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint2_position_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_right_wheel
    pid: {p: 100.0, i: 0.01, d: 10.0}
    
  joint3_position_controller:
    type: velocity_controllers/JointVelocityController
    joint: rear_left_wheel
    pid: {p: 100.0, i: 0.01, d: 10.0}
    
  joint4_position_controller:
    type: velocity_controllers/JointVelocityController
    joint: rear_right_wheel
    pid: {p: 100.0, i: 0.01, d: 10.0}


  
mobile_base_controller:
  type: "diff_drive_controller/DiffDriveController"
  publish_rate: 50

  enable_odom_tf: false  # enable publish on topic /tf

  left_wheel: ['left_front_wheel_joint', 'left_center_wheel_joint', 'left_back_wheel_joint']
  right_wheel: ['right_front_wheel_joint', 'right_center_wheel_joint', 'right_back_wheel_joint']
  # left_wheel: ['left_front_wheel_joint', 'left_back_wheel_joint']
  # right_wheel: ['right_front_wheel_joint', 'right_back_wheel_joint']
  # left_wheel: ['left_back_wheel_joint']
  # right_wheel: ['right_back_wheel_joint']

  wheel_separation: 0.53

  # Odometry covariances for the encoder output of the robot. These values should
  # be tuned to your robot's sample odometry data, but these values are a good place
  # to start
  pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  
  # Top level frame (link) of the robot description
  base_frame_id: base_link

  # Velocity and acceleration limits for the robot
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 6.5     # m/s
      has_acceleration_limits: true
      max_acceleration       : 6.5    # m/s^2
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 1      # rad/s
      has_acceleration_limits: true
      max_acceleration       : 6.0   # rad/s^2
      
controller_list:
  - name: wheels_controller
    action_ns: follow_joint_trajectory
    default: True
    type: FollowJointTrajectory
    joints:
      - front_left_wheel
      - front_right_wheel
      - rear_left_wheel
      - rear_right_wheel


       
